### PR TITLE
[rv_core_ibex] Add testsynthesis target

### PR DESCRIPTION
**Pull request by msfschaffner**
This commit adds basic ASIC synthesis support for Ibex using the DVSIM flow (for quick synthesis experiments).

Signed-off-by: Michael Schaffner <msf@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
LGTM.

File: hw/ip/rv_core_ibex/rv_core_ibex.core [hunk 1]
```diff
@@ -78,4 +78,12 @@ targets:
         verilator_options:
           - "-Wall"
 
+  syn:
+    <<: *default_target
+    # TODO: set default to DC once
+    # this option is available
+    # olofk/edalize#89
```
Comment (imphil) [inline hw/ip/rv_core_ibex/rv_core_ibex.core:85]: Argh, I forgot about that one again. Back on my todo list :-)

File: hw/ip/rv_core_ibex/syn/constraints.sdc [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# set main clock to 125 MHz
+set MAIN_TCK  5.0
+set_ideal_network ${MAIN_CLK_PIN}
+set_ideal_network ${MAIN_RST_PIN}
+set_clock_uncertainty ${SETUP_CLOCK_UNCERTAINTY} ${MAIN_CLK_PIN}
+
+# other timing constraint in ns
+set IN_DEL    1.0
+set OUT_DEL   1.0
+set DELAY     ${MAIN_TCK}
```
Comment (imphil) [inline hw/ip/rv_core_ibex/syn/constraints.sdc:26]: I'd expect nothing spectacular here, but @GregAC can you have a look at those constraints? Are they roughly what you're using in yosys?

Comment (msfschaffner) [reply hw/ip/rv_core_ibex/syn/constraints.sdc:26]: these are just for experiments - we can always adjust them if needed.

**Pull request merged**